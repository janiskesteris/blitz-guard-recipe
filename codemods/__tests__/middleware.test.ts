import { defineSnapshotTest } from "jscodeshift/dist/testUtils"
import fs from "fs"
import { join } from "path"
import MiddlewareTransform from "../middleware"

describe("add middleware", () => {
  defineSnapshotTest(
    MiddlewareTransform,
    null,
    fs
      .readFileSync(join(__dirname, "..", "__testfixtures__/middleware-do-nothing.input.js"))
      .toString()
  )

  defineSnapshotTest(
    MiddlewareTransform,
    null,
    fs.readFileSync(join(__dirname, "..", "__testfixtures__/middleware-basic.input.js")).toString()
  )

  defineSnapshotTest(
    MiddlewareTransform,
    null,
    fs
      .readFileSync(join(__dirname, "..", "__testfixtures__/middleware-with-middleware.input.js"))
      .toString()
  )

  defineSnapshotTest(
    MiddlewareTransform,
    null,
    fs
      .readFileSync(join(__dirname, "..", "__testfixtures__/middleware-with-comments.input.js"))
      .toString()
  )

  defineSnapshotTest(
    MiddlewareTransform,
    null,
    fs
      .readFileSync(join(__dirname, "..", "__testfixtures__/middleware-real-blitz-config.input.js"))
      .toString()
  )
})
