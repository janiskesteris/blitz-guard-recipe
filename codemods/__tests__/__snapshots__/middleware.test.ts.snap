// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add middleware transforms correctly 1`] = `
"const { BlitzGuardMiddleware } = require(\\"@blitz-guard/core/dist/middleware\\")
module.exports = {
  middleware: [
    BlitzGuardMiddleware({
      excluded: [
        \\"/api/auth/mutations/login\\",
        \\"/api/auth/mutations/logout\\",
        \\"/api/guard/queries/getAbility\\",
      ],
    }),
  ],
}"
`;

exports[`add middleware transforms correctly 2`] = `
"const { BlitzGuardMiddleware } = require(\\"@blitz-guard/core/dist/middleware\\");
module.exports = { middleware: [BlitzGuardMiddleware(
  {excluded: [\\"/api/auth/mutations/login\\", \\"/api/auth/mutations/logout\\", \\"/api/guard/queries/getAbility\\"]}
)] }"
`;

exports[`add middleware transforms correctly 3`] = `
"const { BlitzGuardMiddleware } = require(\\"@blitz-guard/core/dist/middleware\\");
const { sessionMiddleware, simpleRolesIsAuthorized } = require(\\"@blitzjs/server\\")
module.exports = { middleware: [
  sessionMiddleware({ isAuthorized: simpleRolesIsAuthorized }),
  BlitzGuardMiddleware(
    {excluded: [\\"/api/auth/mutations/login\\", \\"/api/auth/mutations/logout\\", \\"/api/guard/queries/getAbility\\"]}
  )
] }"
`;

exports[`add middleware transforms correctly 4`] = `
"const { BlitzGuardMiddleware } = require(\\"@blitz-guard/core/dist/middleware\\");
const { sessionMiddleware, simpleRolesIsAuthorized } = require(\\"@blitzjs/server\\")
module.exports = {
  middleware: [
    sessionMiddleware({ isAuthorized: simpleRolesIsAuthorized }),
    BlitzGuardMiddleware(
      {excluded: [\\"/api/auth/mutations/login\\", \\"/api/auth/mutations/logout\\", \\"/api/guard/queries/getAbility\\"]}
    )
  ],

  /* Uncomment this to customize the webpack config
webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
// Note: we provide webpack above so you should not \`require\` it
// Perform customizations to webpack config
// Important: return the modified config
return config
},
  */
}"
`;

exports[`add middleware transforms correctly 5`] = `
"const { BlitzGuardMiddleware } = require(\\"@blitz-guard/core/dist/middleware\\");
const { sessionMiddleware, simpleRolesIsAuthorized } = require(\\"@blitzjs/server\\")
const withBundleAnalyzer = require(\\"@next/bundle-analyzer\\")({
  enabled: process.env.ANALYZE === \\"true\\",
})

module.exports = withBundleAnalyzer({
  middleware: [sessionMiddleware({
    isAuthorized: simpleRolesIsAuthorized,
    sessionExpiryMinutes: 4,
  }), BlitzGuardMiddleware(
    {excluded: [\\"/api/auth/mutations/login\\", \\"/api/auth/mutations/logout\\", \\"/api/guard/queries/getAbility\\"]}
  )],
  log: {
    // level: \\"trace\\",
  },
  experimental: {
    isomorphicResolverImports: false,
  },
  /*
  webpack: (config, {buildId, dev, isServer, defaultLoaders, webpack}) => {
    // Note: we provide webpack above so you should not \`require\` it
    // Perform customizations to webpack config
    // Important: return the modified config
    return config
  },
  webpackDevMiddleware: (config) => {
    // Perform customizations to webpack dev middleware config
    // Important: return the modified config
    return config
  },
  */
})"
`;
